### PR TITLE
TIP-1233: upgrade to PHP 7.4

### PR DESCRIPTION
**Pull request by jjanvier**
Check https://github.com/akeneo/pim-enterprise-dev/pull/8092 for the description and to know how to merge it.

### CODE REVIEW

**Comment by lpalgarvio** [comment]
skipped deployment issues on php 7.4 by using `composer create-project --ignore-platform-reqs akeneo/pim-community-standard ...` but didnt test yet make and etc.

**Comment by jjanvier** [comment]
Hello @lpalgarvio 

To know if PHP's minor version is a strong requirement or not, I need to be able to launch our CI workflows on Community and Enterprise editions.

**Comment by jjanvier** [comment]
:elephant: :seven: :diamond_shape_with_a_dot_inside: :four:  :crossed_fingers: :heart:

**Review by nmarniesse — APPROVED** [review]

File: src/Akeneo/Pim/Enrichment/Component/Product/Comparator/Attribute/MetricComparator.php [hunk 1]
```diff
@@ -47,7 +47,7 @@ public function compare($data, $originals)
             return null;
         }
 
-        if (null === $data['data']['amount']) {
+        if (null === $data['data'] || null === $data['data']['amount']) {
```
Comment (mmetayer) [inline src/Akeneo/Pim/Enrichment/Component/Product/Comparator/Attribute/MetricComparator.php:50]:
```suggestion
        if (!isset($data['data']['amount'])) {
```

**Review by mmetayer — APPROVED** [review]

**Review by ValentinMumble — APPROVED** [review]

**Comment by jjanvier** [comment]
\o/

**Comment by lpalgarvio** [comment]
Hello everyone,

thanks for the development here. I see that incompatibilities for PHP 7.4 were indeed found and already merged to master.
https://github.com/akeneo/pim-community-dev/blob/master/CHANGELOG.md

Any idea when this will be merged to a new release in v4.0.x?

It's not documented in v4 changelog yet and it doesn't show up yet in commits for latest v4.0.x versions
https://github.com/akeneo/pim-community-dev/blob/v4.0.66/CHANGELOG-4.0.md
https://github.com/akeneo/pim-community-dev/commits/v4.0.66

We currently have an issue with exporting products to xlsx... the jobs get stuck forever and we are flooded with PHP deprecation warnings.

Thanks!

**Comment by jjanvier** [comment]
@lpalgarvio as it changes the system requirements of the software, it probably won't be backported to 4.0. Next PIM version will be compatible with PHP 7.4, not the current one. But I let the team answer you.

**Comment by bartoll** [comment]
Could someone write here when the next version of PIM is released? Is it a long wait? I mean, of course, the version with PHP 7.4 support.

**Comment by nmarniesse** [comment]
Hello, the next PIM release (5.0) will be available in early 2021. I can confirm it will support php 7.4

**Pull request merged**